stlab/concurrency/channel.hpp

```diff
@@ -691,7 +691,7 @@ struct shared_process : shared_process_receiver<R>,
     bool                     _process_close_queue = false;
     // REVISIT (sparent) : I'm not certain final needs to be under the mutex
     bool                     _process_final = false;
-    std::shared_ptr<std::function<void()>> _process_timeout_function;
+    std::atomic_size_t       _process_timeout_control{0};
```

Comment:
This appears to have values 0, 1, and 2 - with some assigned meaning. What is the meaning? Maybe make an atomic<enum>?